cluster/shard_writer.go

```diff
@@ -61,9 +62,20 @@ func (w *ShardWriter) WriteShard(shardID, ownerID uint64, points []models.Point)
 		conn.Close() // return to pool
 	}(conn)
 
+	// Determine the location of this shard and whether it still exists
+	db, rp, sgi := w.MetaClient.ShardOwner(shardID)
```

Comment:
No.  The only way this code could get executed is if the `PointsWriter` has succeeded (on the same node) and created the shards in the meta store already.